modules/models/record.php

```diff
@@ -210,6 +211,52 @@ public function save()
             }
         }
     }
+    /*
+        I decided to make this a private function that is only invoked internally by save.
+        Save can be considered to be "saving details in the database" so the user shouldnt
+        save a dirty record differently to how they save a clean record.
+    */
+    private function update($pk='id')
+    {
+        // Get attributes of the called object
+        $attrs = $this->getAttrs();
+
+        // Ensure that the pk exists
+        if (!in_array($pk, $attrs)) {
+            throw new ErrorException("Record trying to delete using a primary key that does not exist: ".$pk);
+        }
+
+        // Get the record with the primary key from the table
+        $old = get_called_class()::find_by(array($pk => $this->$pk));
```

Comment:
Ok I guess we could do this for now, an alternative way would be to note down which specific columns are dirty but I'm cool with leaving this as an optimisation for some other time

Reply:
yeah I was thinking the same, but the solution is a little bit more involved. It becomes complicated since the user is allowed to directly change the attributes of the record. In the future if we have time, we can make the attributes private and force the user to go through get and set, then we can track which columns are dirty